pandas/core/indexes/category.py

```diff
@@ -661,20 +661,58 @@ def is_dtype_equal(self, other):
     take_nd = take
 
     def map(self, mapper):
-        """Apply mapper function to its categories (not codes).
+        """
+        Map index values using input correspondence (a dict, Series, or
+        function).
+
+        Maps the values (their categories, not the codes) of the index to new
+        categories. If the mapping correspondence maps each original category
+        to a different new category the result is a CategoricalIndex which has
+        the same order property as the original, otherwise an Index is
+        returned.
+
+        If a dictionary or Series is used any unmapped category is mapped to
+        NA. Note that if this happens an Index will be returned.
 
         Parameters
         ----------
-        mapper : callable
-            Function to be applied. When all categories are mapped
-            to different categories, the result will be a CategoricalIndex
-            which has the same order property as the original. Otherwise,
-            the result will be a Index.
+        mapper : function, dict, or Series
+            Mapping correspondence.
 
         Returns
         -------
-        applied : CategoricalIndex or Index
+        CategoricalIndex or Index
+            Mapped index.
 
+        See Also
+        --------
+        Index.map : Apply a mapping correspondence on an Index.
+        Series.map : Apply a mapping correspondence on a Series.
+        Series.apply : Apply more complex functions on a Series.
+
+        Examples
```

Comment:
Good to see Examples here - why not add to other `map` implementations as well?

Reply:
This is outside the original scope of the contribution, see first PR comment.